### PR TITLE
Fix subparametric elements in MixedDofHandler

### PR DESCRIPTION
**Pull request by koehlerson**
Fixed https://github.com/KristofferC/JuAFEM.jl/issues/315 and adds regression tests

### CODE REVIEW

**Comment by lijas** [comment]
Nice thanks.

In the future it could be nice to make the FieldHandler be associated with a cell-type. That way we get the geometric interpolation "automatically" (via `default_interpolation(celltype)`).
One could also then check that the added fields are compatible with the cell type (so that one does not add a triangular interpolation to a Quadrilateral for example). 

```
fh = FieldHandler(Cell{2,4,4})
push!(fh, Field(:u, Lagrange{2,RefCube,2})) #works
push!(fh, Field(:u, Lagrange{2,RefTetrahedron,2})) #error or warning
```

File: src/Dofs/MixedDofHandler.jl [hunk 1]
```diff
@@ -9,9 +9,9 @@ end
 
 Construct a `FieldHandler` based on an array of `Field`s and assigns it a set of cells.
 """
-function FieldHandler(fields::Vector{Field}, cellset)
+function FieldHandler(fields::Vector{Field}, cellset, ipgeo::Vector{ip}=getproperty.(fields,:interpolation)) where ip <: Interpolation
```
Comment (kimauth) [inline src/Dofs/MixedDofHandler.jl:12]: Geometry interpolation is called `geom_interpol` in most of the files, maybe be consistent with that?

File: src/Dofs/MixedDofHandler.jl [hunk 2]
```diff
@@ -9,9 +9,9 @@ end
...
     # TODO for now, only accept isoparamtric mapping
```
Comment (kimauth) [inline src/Dofs/MixedDofHandler.jl:13]:
```suggestion

```

**Review by kimauth — COMMENTED** [review]
We should add a test for the dof distribution of sub/superparametric elements. (I think it should work fine though.)
Otherwise this looks good to be merged to me. :)

**Comment by kimauth** [comment]
> In the future it could be nice to make the FieldHandler be associated with a cell-type. That way we get the geometric interpolation "automatically" (via `default_interpolation(celltype)`).
> One could also then check that the added fields are compatible with the cell type (so that one does not add a triangular interpolation to a Quadrilateral for example).
> 
> ```
> fh = FieldHandler(Cell{2,4,4})
> push!(fh, Field(:u, Lagrange{2,RefCube,2})) #works
> push!(fh, Field(:u, Lagrange{2,RefTetrahedron,2})) #error or warning
> ```

With [#320 ](https://github.com/KristofferC/JuAFEM.jl/pull/320) we at least check this now when the FieldHandler is added to the MixedDofHandler, so the user does get warned when using an incompatible interpolation on a cellset.

**Comment by koehlerson** [comment]
Should we close this? As far as I understand it's not longer needed since `FieldHandler` got rid of the `BCValues`?

**Comment by kimauth** [comment]
Does that mean #315 is resolved? Then we should close that one, too.

**Comment by lijas** [comment]
I think so, but I have not tested if 315 though.

**Comment by koehlerson** [comment]
Ok, so I can update the regression tests of this branch, such that #315 can be closed

**Pull request closed without merging**